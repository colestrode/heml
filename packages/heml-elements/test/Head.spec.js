const render = require('@heml/render')
const Head = require('../build/Head').default
const Style = require('../build/Style').default
const Meta = require('../build/Meta').default

beforeAll(() => {
  Style.preRender({ elements: [] });
  Meta.preRender();
})

test('renders with some attrs', () => {
  const attrs = {
    margin: '1px 2px 3px 4px'
  }

  return render.renderElement(Head, attrs, '<contents></contents>')
    .then((res) => expect(res).toMatchSnapshot())
});
