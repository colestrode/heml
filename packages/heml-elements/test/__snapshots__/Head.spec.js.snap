// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with some attrs 1`] = `
"<head> </head><head margin=\\"1px 2px 3px 4px\\"><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\" /><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" /><meta name=\\"x-apple-disable-message-reformatting\\" /><!--[if !mso]><!--><meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" /><!--<![endif]--><style type=\\"text/css\\" data-embed>
        * { text-size-adjust: 100%; -ms-text-size-adjust: 100%; -moz-text-size-adjust: 100%; -webkit-text-size-adjust: 100%; }
        html { height: 100%; width: 100%; }
        body { height: 100% !important; margin: 0 !important; padding: 0 !important; width: 100% !important; mso-line-height-rule: exactly; }
        div[style*=\\"margin: 16px 0\\"] { margin:0 !important; }
        table, td { mso-table-lspace: 0pt; mso-table-rspace: 0pt; }
        img { border: 0; height: auto; line-height: 100%; outline: none; text-decoration: none; -ms-interpolation-mode: bicubic;}
        .ReadMsgBody, .ExternalClass { width: 100%; }
        .ExternalClass, .ExternalClass p, .ExternalClass span, .ExternalClass td, .ExternalClass div { line-height: 100%; }
      </style><style type=\\"text/css\\">
        h1, h2, h3, h4, h5, h6 { margin: 20px 0; }
        h1 { line-height: 40px; }
        h2 { line-height: 30px; }
        h3 { line-height: 24px; }
        h5 { line-height: 17px; }
        h6 { line-height: 12px; }
        p { display: block; margin: 14px 0; }
        ul { margin-left: 20px; margin-top: 16px; margin-bottom: 16px; padding: 0; list-style-type: disc; }
      </style><!--[if gte mso 9]>
      <style type=\\"text/css\\">
      li { text-indent: -1em; }
      table td { border-collapse: collapse; }
      </style>
      <![endif]--><title> </title></style><!-- content --><contents></contents><!--[if gte mso 9]><xml>
       <o:OfficeDocumentSettings>
        <o:AllowPNG/>
        <o:PixelsPerInch>96</o:PixelsPerInch>
       </o:OfficeDocumentSettings>
      </xml><![endif]--></head>"
`;
